src/js/components/Box/StyledBox.js

```diff
@@ -211,6 +211,34 @@ const gapStyle = (directionProp, gap, responsive, wrap, theme) => {
     if (responsiveMetric) {
```

Comment:
Should the check of if `typeof gap === 'object'` come before this `if (directionProp === 'column' || directionProp === 'column-reverse')` block?